src/openforms/submissions/api/validation.py

```diff
@@ -105,10 +105,13 @@ def get_invalid_prefilled_fields(
 
         plugin_name = component["prefill"]["plugin"]
         attribute_name = component["prefill"]["attribute"]
-        if (
-            submission_step.data[component["key"]]
-            != submission.prefill_data[plugin_name][attribute_name]
-        ):
+
+        if submission.prefill_data[plugin_name][attribute_name] is None:
+            initial_value = component["defaultValue"]
```

Comment:
I think here we need to keep the old code path if `submission.prefill_data[plugin_name][attribute_name]` is not `None`, but if it is `None` we could check that `submission_step.data[component["key"]] == component["defaultValue"]` (and if they are different add the component label to the `invalid_prefilled_fields`).

On the other hand, I was thinking that there may be issues with the default values (Formio does some funny things with default values), so I'm still not 100% sure we should do this extra check :thinking:

Reply:
Oups, didn't see Sergei had already reviewed! :sweat_smile:

Reply:
Yes ahah :smile: If defaultValue cannot be trusted, I think we should go for the Sergei's solution and keeping things simple for now then